### PR TITLE
Add User-Agent header to HTTPRequest

### PR DESCRIPTION
**Pull request by kunall17**
Implement #111 
Adds header like ZulipAndroid-0.8.9 
And upgrade okHttp library dependency

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@kunall17, it looks like you've already signed the Dropbox CLA. Thanks!

File: app/src/main/java/com/zulip/android/ZulipApp.java [hunk 1]
```diff
@@ -158,12 +158,12 @@ public String getApiKey() {
     public String getUserAgent() {
         try {
             return ZulipApp.USER_AGENT
-                    + "/"
+                    + "-"
```
Comment (niftynei) [inline app/src/main/java/com/zulip/android/ZulipApp.java]: This should stay as a `/`.  User agents are usually in the form `software/version`

File: app/src/main/java/com/zulip/android/ZulipApp.java [hunk 2]
```diff
@@ -158,12 +158,12 @@ public String getApiKey() {
...
                     + getPackageManager().getPackageInfo(getPackageName(), 0).versionName;
         } catch (NameNotFoundException e) {
             // This should… never happen, but okay.
             ZLog.logException(e);
-            return ZulipApp.USER_AGENT + "/unknown";
+            return ZulipApp.USER_AGENT + "-unknown";
```
Comment (niftynei) [inline app/src/main/java/com/zulip/android/ZulipApp.java]: Same as above comment. This should stay as a slash.

**Comment by niftynei** [comment]
See comments about `/` vs `-`.  

Otherwise lgtm

**Comment by niftynei** [comment]
Thanks @kunall17 !

**Pull request merged**